BinaryTreeNodeTest.java

```diff
@@ -0,0 +1,76 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BinaryTreeNodeTest {
+    private BinaryTreeNode<Integer> n1, n2, n3, n4, n5, n6, n7, n8;
+
+    @BeforeEach
+    void setUp() {
+        // Set up the tree
+        n1 = new BinaryTreeNode<>(1, null, null);
+        n2 = new BinaryTreeNode<>(5, null, null);
+        n3 = new BinaryTreeNode<>(3, n1, n2);
+        n4 = new BinaryTreeNode<>(14, null, null);
+        n5 = new BinaryTreeNode<>(9, n3, n4);
+        n6 = new BinaryTreeNode<>(19, null, null);
+        n7 = new BinaryTreeNode<>(18, null, n6);
+        n8 = new BinaryTreeNode<>(16, n5, n7);
+    }
+
+    @Test
+    void findAncestorsTest() {
+        // Normal case
+        assertEquals(n8.findAncestors(18), Collections.singletonList(16));
```

Comment:
After giving it a second thought you are right. `singletonList` is OK here.
It is just quite rarely used method (I don't think I have ever used it or seen it in production code) since it is very, very specialized (immutable list of one element rather than something more general), so it felt a bit odd.

Google code has a strong preference towards things like [ImmutableList.of(...)](http://google.github.io/guava/releases/22.0/api/docs/com/google/common/collect/ImmutableList.html#of-E-) from Guava, which is more general (general purpose immutable list, just created via convenience method taking a single argument, rather than special method to create single-element list) but that would add another dependency to the project just to provide one class.